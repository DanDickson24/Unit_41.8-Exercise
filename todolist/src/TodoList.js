import React, { useState } from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './ToDo';

function TodoList() {
  const [todos, setTodos] = useState([]);

  // add a new todo
  const addTodo = (newTodo) => {
    setTodos([...todos, newTodo]);
  };

  // remove a specific todo by index
  const removeTodo = (index) => {
    const newTodos = [...todos];
    newTodos.splice(index, 1);
    setTodos(newTodos);
  };

  // edit a specific todo by index and update the task
  const editTodo = (index, editedTask) => {
    const newTodos = [...todos];
    newTodos[index] = editedTask;
    setTodos(newTodos);
  };

  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      <ul>
        {todos.map((todo, index) => (
          <Todo
            key={index}
            index={index}
            task={todo}
            removeTodo={removeTodo}
            editTodo={editTodo}
          />
        ))}
      </ul>
    </div>
  );
}

export default TodoList;
