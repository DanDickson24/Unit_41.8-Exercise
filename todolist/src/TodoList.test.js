import React from 'react';
import { render } from '@testing-library/react';
import TodoList from './TodoList';

test('TodoList snapshot test', () => {
  const { container } = render(<TodoList />);
  expect(container).toMatchSnapshot();
});

test('TodoList smoke test', () => {
    const { getByText } = render(<TodoList />);
    expect(getByText('Todo List')).toBeInTheDocument();
  });