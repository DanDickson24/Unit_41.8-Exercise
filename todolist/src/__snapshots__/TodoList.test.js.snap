// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList snapshot test 1`] = `
<div>
  <div>
    <form>
      <input
        placeholder="Enter a task"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <ul />
  </div>
</div>
`;
